File: demo/src/pages/ExplicitTransition/ExplicitTransition.ts
import {main, Viewable,State,Button,withAnimation,Text, Bound, Animation, AnimationTool} from "@tswift/ui";

@main
export class ExplicitTransition extends Viewable {
    @State showBanana = false
    
     body = ()=>[
        Button("Toggle",()=>
            withAnimation('.easeIn', ()=> this.showBanana = !this.showBanana )
        ),
        this.showBanana && Text("🍌")
                .transition('.slide') 
    ]
}

File: demo/src/pages/ExplicitTransition/ExplicitTransition.swift
//
//  ExplicitTransitionView.swift
//  playground
//
//  Created by Justin Spears on 9/15/22.
//
// [from](https://swiftontap.com/withanimation(_:_:))

import SwiftUI
struct ExplicitTransition: View {
    @State var showBanana = false
    
    var body: some View {
        Button("Toggle") {
            withAnimation(.easeInOut) { showBanana.toggle() }
        }.padding()
        if showBanana {
            Text("🍌")
                .padding()
                .transition(.slide)
        }
    }
    
}
